validator_tmparchive.rb

```diff
@@ -22,46 +22,65 @@
 
 #---------------------  METHODS
 def set_submitter_info(logger,user_email,user_name,contacts_hash,status_hash)
-	if user_email == ''
-	    status_hash['api_ok'] = false
-	    contacts_hash.merge!(submitter_name: 'Workflows')
-	    contacts_hash.merge!(submitter_email: 'workflows@macmillan.com')
-	    logger.info {"dropbox api may have failed, not finding file metadata"}
-	else
-		#check to see if submitter is in ebooks dept.:
-		staff_hash = Mcmlln::Tools.readjson(Val::Files.staff_emails)  		#read in our static pe/pm json
-		for i in 0..staff_hash.length - 1
-				if user_email.downcase == "#{staff_hash[i]['email'].downcase}"
-						if "#{staff_hash[i]['division']}" == 'Ebooks' || "#{staff_hash[i]['division']}" == 'Workflow'
-								contacts_hash['ebooksDept_submitter'] = true
-								logger.info {"#{user_name} is a member of ebooks or Workflow dept, flagging that to edit user comm. addressees"}
-						end
-				end
-		end
+  if user_email == ''
+    status_hash['api_ok'] = false
+    contacts_hash.merge!(submitter_name: 'Workflows')
+    contacts_hash.merge!(submitter_email: 'workflows@macmillan.com')
+    logger.info {"dropbox api may have failed, not finding file metadata"}
+  else
+    #check to see if submitter is in ebooks dept.:
+    staff_hash = Mcmlln::Tools.readjson(Val::Files.staff_emails)  		#read in our static pe/pm json
+    for i in 0..staff_hash.length - 1
+      if user_email.downcase == "#{staff_hash[i]['email'].downcase}"
+        if "#{staff_hash[i]['division']}" == 'Ebooks' || "#{staff_hash[i]['division']}" == 'Workflow'
+          contacts_hash['ebooksDept_submitter'] = true
+          logger.info {"#{user_name} is a member of ebooks or Workflow dept, flagging that to edit user comm. addressees"}
+        end
+      end
+    end
     #writing user info from Dropbox API to json
     contacts_hash.merge!(submitter_name: user_name)
     contacts_hash.merge!(submitter_email: user_email)
     Vldtr::Tools.write_json(contacts_hash,Val::Files.contacts_file)
     logger.info {"file submitter retrieved, display name: \"#{user_name}\", email: \"#{user_email}\", wrote to contacts.json"}
-	end
+  end
 end
 def nondoc(logger,status_hash)
-	status_hash['docfile'] = false
-	logger.info {"This is not a .doc or .docx file. Posting error.txt to the project_dir for user."}
-	File.open(Val::Files.errFile, 'w') { |f|
-			f.puts "Unable to process \"#{Val::Doc.filename_normalized}\". Your document is not a .doc or .docx file."
-	}
+  status_hash['docfile'] = false
+  logger.info {"This is not a .doc or .docx file. Posting error.txt to the project_dir for user."}
+  File.open(Val::Files.errFile, 'w') { |f|
+    f.puts "Unable to process \"#{Val::Doc.filename_normalized}\". Your document is not a .doc or .docx file."
+  }
+end
+def convertDocToDocxPSscript(logger, doc_or_docx_workingfile)
+  saveas_output = ''
```

Comment:
I had some circular login there, good catch.  I didn't need that var at all, since it was just for the rescue, and if we needed the rescue, the var wouldn't have any value.  So I ditched it:)